cmd/swarm/config_test.go

```diff
@@ -56,15 +60,162 @@ func TestConfigFailsSwapEnabledNoSwapApi(t *testing.T) {
 	swarm.ExpectExit()
 }
 
-func TestConfigFailsNoBzzAccount(t *testing.T) {
+func TestBzzKeyFlag(t *testing.T) {
+	key, err := crypto.GenerateKey()
+	if err != nil {
+		t.Fatal("unable to generate key")
+	}
+	hexKey := hex.EncodeToString(crypto.FromECDSA(key))
+	bzzaccount := crypto.PubkeyToAddress(key.PublicKey).Hex()
+
+	dir, err := ioutil.TempDir("", "bzztest")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(dir)
+
+	conf := &node.Config{
+		DataDir: dir,
+		IPCPath: fmt.Sprintf("bzzd-%s.ipc", bzzaccount),
+		NoUSB:   true,
+	}
+
+	node := &testNode{Dir: dir}
+
 	flags := []string{
 		fmt.Sprintf("--%s", SwarmNetworkIdFlag.Name), "42",
-		fmt.Sprintf("--%s", SwarmPortFlag.Name), "54545",
+		fmt.Sprintf("--%s", SwarmPortFlag.Name), "0",
+		fmt.Sprintf("--%s", utils.ListenPortFlag.Name), "0",
+		fmt.Sprintf("--%s", utils.DataDirFlag.Name), dir,
+		fmt.Sprintf("--%s", utils.IPCPathFlag.Name), conf.IPCPath,
+		fmt.Sprintf("--%s", SwarmBzzKeyHexFlag.Name), hexKey,
 	}
 
-	swarm := runSwarm(t, flags...)
-	swarm.Expect("Fatal: " + SwarmErrNoBZZAccount + "\n")
-	swarm.ExpectExit()
+	node.Cmd = runSwarm(t, flags...)
+	defer func() {
+		node.Shutdown()
+	}()
+
+	node.Cmd.InputLine(testPassphrase)
+
+	// wait for the node to start
+	for start := time.Now(); time.Since(start) < 10*time.Second; time.Sleep(50 * time.Millisecond) {
+		node.Client, err = rpc.Dial(conf.IPCEndpoint())
+		if err == nil {
+			break
+		}
+	}
+	if node.Client == nil {
+		t.Fatal(err)
+	}
+
+	// load info
+	var info swarm.Info
+	if err := node.Client.Call(&info, "bzz_info"); err != nil {
+		t.Fatal(err)
+	}
+
+	if info.BzzAccount != bzzaccount {
+		t.Fatalf("Expected account to be %s, got %s", bzzaccount, info.BzzAccount)
+	}
+
```

Comment:
Please no need for new line at 121, but we need a new line between the two tests (line 122:123)